client: Unified WithPlatform option

vvoland opened the pull request:
- stacked on: https://github.com/moby/moby/pull/50930
- related to: https://github.com/moby/moby/issues/50927
- fixes: https://github.com/moby/moby/issues/50967

Use a single `WithPlatform` functional option for ImageInspect and
ImageHistory which both can accept a single platform to operate on.

The non-unified functional opts returning the same type are neatly grouped in GoDoc:

<img width="592" height="87" alt="image" src="https://github.com/user-attachments/assets/da57aaa7-05fc-4e8f-b38c-ec1c5e7cf130" />


the only issue are with the unified options like `WithPlatform`:

<img width="664" height="404" alt="image" src="https://github.com/user-attachments/assets/550e4684-87f6-41b7-be2a-d76c0ad821d8" />

thaJeztah commented:
Tricky one; implementation-wise this looks like the most suitable approach, but the downside looks to be that it's not possible to discover which options can be used for each; at least `pkg.go.dev` is not smart enough to tie these to a specific Option-type, so `WithPlatform` is neither listed under `ImageInspectOption`  nor under `ImageHistoryOption`, and it only shows up under "functions";

<img width="635" height="102" alt="Screenshot 2025-09-09 at 22 16 54" src="https://github.com/user-attachments/assets/fe246510-6a9e-4ee4-a9fe-7049a0cdaf15" />
<img width="308" height="105" alt="Screenshot 2025-09-09 at 22 17 17" src="https://github.com/user-attachments/assets/b1b3443a-a2a5-4f42-b205-ede7d0370377" />
<img width="239" height="50" alt="Screenshot 2025-09-09 at 22 18 35" src="https://github.com/user-attachments/assets/408a617b-c5e1-40ab-af0a-165b6a58f104" />

vvoland commented on client/image_history_opts.go:
```diff
@@ -2,16 +2,9 @@ package client
 
-// ImageHistoryOption is a type representing functional options for the image history operation.
 type ImageHistoryOption interface {
-	Apply(*imageHistoryOpts) error
+	ApplyImageHistoryOption(*imageHistoryOpts) error
```
For some reason I was convinced that the methods of the interface must be exported, but looks like it's actually not the case.
Yeah in that case it definitely should be non-exported!

vvoland committed:
client: Unified WithPlatform option

Use a single `WithPlatform` functional option for `ImageInspect` and
`ImageHistory` which both can accept a single platform to operate on.

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
client: Change `ImageHistory` functional opts

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
client: Migrate `ImageInspect` to functional opts

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>
